src/main/java/at/ac/tuwien/kr/alpha/grounder/transformation/SumNormalization.java

```diff
@@ -0,0 +1,169 @@
+		ArrayList<Rule> additionalRules = new ArrayList<>();
+
+		for (Iterator<Literal> iterator = rule.getBody().iterator(); iterator.hasNext();) {
+			Literal bodyElement = iterator.next();
+			// Skip non-aggregates.
+			if (!(bodyElement instanceof AggregateLiteral)) {
+				continue;
+			}
+			AggregateLiteral aggregateLiteral = (AggregateLiteral) bodyElement;
+			AggregateAtom aggregateAtom = aggregateLiteral.getAtom();
+
+			// FIXME: limited rewriting of lower-bounded cardinality/sum aggregates only.
+			if (aggregateLiteral.isNegated() || aggregateAtom.getUpperBoundOperator() != null
+				|| (aggregateAtom.getAggregatefunction() != AggregateAtom.AGGREGATEFUNCTION.COUNT
+					&& aggregateAtom.getAggregatefunction() != AggregateAtom.AGGREGATEFUNCTION.SUM)
+				|| aggregatesInRule++ > 0) {
+				throw new UnsupportedOperationException("Only limited #count/#sum aggregates without upper bound are currently supported." +
+					"No rule may have more than one aggregate.");
+			}
+
+			// Only treat sum aggregates.
+			if (aggregateAtom.getAggregatefunction() != AggregateAtom.AGGREGATEFUNCTION.SUM) {
+				continue;
+			}
+			// Remove aggregate from rule body.
+			iterator.remove();
+
+			// Prepare aggregate parameters.
+			aggregateCount++;
+			Substitution aggregateSubstitution = new Substitution();
+			aggregateSubstitution.put(VariableTerm.getInstance("AGGREGATE_ID"), ConstantTerm.getInstance(aggregateCount));
+			aggregateSubstitution.put(VariableTerm.getInstance("LOWER_BOUND"), aggregateAtom.getLowerBoundTerm());
+
+			// Create new output atom for addition to rule body instead of the aggregate.
+			aggregateOutputAtoms.add(aggregateOutputAtom.substitute(aggregateSubstitution).toLiteral());
+
+			// Create input to sorting network from aggregate elements.
+			for (AggregateAtom.AggregateElement aggregateElement : aggregateAtom.getAggregateElements()) {
+				// Prepare element substitution.
+				List<Term> elementTerms = aggregateElement.getElementTerms();
+				FunctionTerm elementTuple = FunctionTerm.getInstance("element_tuple", elementTerms);
+				Substitution elementSubstitution = new Substitution(aggregateSubstitution);
+				elementSubstitution.put(VariableTerm.getInstance("ELEMENT_TUPLE"), elementTuple);
+				elementSubstitution.put(VariableTerm.getInstance("FIRST_VARIABLE"), elementTuple.getTerms().get(0));
+
+				// Create new rule for input.
+				BasicAtom inputHeadAtom = aggregateInputAtom.substitute(elementSubstitution);
+				List<Literal> elementLiterals = new ArrayList<>(aggregateElement.getElementLiterals());
+				Rule inputRule = new Rule(new DisjunctiveHead(Collections.singletonList(inputHeadAtom)), elementLiterals);
+				additionalRules.add(inputRule);
+			}
+
+			// Create lower bound for the aggregate.
+			BasicAtom lowerBoundHeadAtom = lowerBoundAtom.substitute(aggregateSubstitution);
+			// FIXME: we need a deep copy of the rule body here (as otherwise, e.g., interval atoms are shared and their later rewriting fails).
```

Comment:
Please either resolve remaining inline TODOs before merging to master or create issues to track them.